main.js

```diff
@@ -0,0 +1,179 @@
+}
+
+/**
+ * Exerciess 4: 
+ * Hàm nhận tham số là n (số nguyên dương), 1 <= n <= 30 
+ * In ra n! (giai thừa của n)
+ */
+
+const calculateFactorial = num => {
+    let factorial = 1
+
+    if ( 1 <= num <= 30 ) {
+        for (let i = 1; i <= num; i++) {
+            factorial *= i
+        }
+    }  
+    return console.log('Giai thừa của', num, 'là', factorial)  
+}
+
+/**
+ * Exerciess 5: 
+ * Hàm nhận vào 1 mảng đếm xem trong mảng có bao nhiêu số chẵn
+ * Vd: param là [1, 2, 3, 6, 2, 9, 0] thì return 4 vì có 4 số chẵn
+ */
+
+const getEvenNum = () => {
+    let evenNum = 0
+    array.forEach(num => {
+      if (num % 2 === 0) {
+        evenNum++
+        }
+    })
+    return console.log('có:', evenNum, 'số chẵn');
+}
+
+console.log('Check Ex_5');
+array = [1,4,56,78,43,23,45]
+getEvenNum (array)
+
+/**
+ * Exerciess 6: 
+ * Hàm nhận vào 1 mảng chứa các chữ cái bị trùng lặp ngẫu nhiên 
+ * Tính toán và trả về 1 mảng đã xóa đi các chữ cái bị trùng lặp
+ */
+const deLetter = array => {
+    const getLetter = [];
+    for (let i = 0; i < array.length; i++) {
+        if (getLetter.indexOf(array[i]) === -1) {
+            getLetter.push(array[i])
+        }
+    }
+    return console.log (getLetter)
+}
+
+
+console.log('Check Ex_6');
+getLetter = ['A', 'C', 'A', 'A', 'B', 'D', 'B']
+deLetter (getLetter)
+
+ /**
+ * Exerciess 7: 
+ * Bạn cần phải tạo ra 1 object có đầy đủ các thuộc tính: name, score từ 2 array
+ */
+
+const studentNames = [
+    { id: 1, name: 'Nguyễn Thị Tèo' },
+    { id: 2, name: 'Phạm Văn Bưởi' },
+    { id: 3, name: 'Hoàng Văn Nam' },
+    { id: 4, name: 'Vũ Ngọc Duy' },
+    { id: 5, name: 'Nguyễn Minh Nhật' },
+    { id: 6, name: 'Phí Duy Quân' },
+    { id: 7, name: 'Trần Minh Minh' }
+]
+  const studentScores = [
+    { id: 1, score: 9.2 },
+    { id: 2, score: 2.3 },
+    { id: 3, score: 3.7 },
+    { id: 4, score: 6.9 },
+    { id: 5, score: 5.2 },
+    { id: 6, score: 9.6 },
+    { id: 7, score: 6.1 }
+]
+const listStudent = () => {
+    for (let i = 0 ; i < studentNames.length ; i++ ) {
+      for (let j = 0 ; j < studentScores.length ; j ++) {
+        if (studentNames[i].id === studentScores[j].id) {
+          studentNames[i].score  = studentScores[j].score
+        }
+      }
+    }
+    return console.log(studentNames);
+  }
+console.log('Bảng danh sách học sinh và điểm số');
+listStudent ()
+
+/**
+ * Exerciss 8:
+ * Tìm sinh viên có điểm (score) thấp nhất và sinh viên có score cao nhất 
+ */
+
+const students = [
+    { id: 1, name: 'Nguyễn Thị Tèo', score: 9.2 },
+    { id: 2, name: 'Phạm Văn Bưởi', score: 2.3 },
+    { id: 3, name: 'Hoàng Văn Nam', score: 3.7 },
+    { id: 4, name: 'Vũ Ngọc Duy', score: 6.9 },
+    { id: 5, name: 'Nguyễn Minh Nhật', score: 5.2 },
+    { id: 6, name: 'Phí Duy Quân', score: 9.6 },
+    { id: 7, name: 'Trần Minh Minh', score: 6.1 }
+]
+const scoreMaxMin = () => {
+    let scoreMax = 0 
+    let scoreMin = 12
+    let theBest = {}
+    let theBad = {}
```

Comment:
```suggestion
    const theBest = []
    const theBad = []
```
1: vì 2 biến này ko bị gán lại nên phải đặt là const
2. phải đặt là array thay vì object vì cũng 1 lúc có thể có nhiều người cao điểm nhất (vd cả 2 cùng 10 điểm), và cùng 1 lúc có thể có nhiều người thấp điểm nhất (vd cả lớp đều 0 điểm)